packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -92,3 +95,73 @@ export function renderMeshOwner(resourceAccount: string, meshAccount: string) :
     ? meshAccount
     : undefined;
 }
+
+/**
+ * This is the helper method to check if properties for passed match are undefined.
+ * This returns true if all properties for an object are undefined.
+ */
+export function areMatchPropertiesUndefined(match: HttpRouteMatch | GrpcRouteMatch | GrpcGatewayRouteMatch | undefined): boolean {
+  let isEmpty = true;
+  for (const property in match) {
+    if (property && (property !== 'rewriteRequestHostname' && property !== 'pathRewrite')) {
+      isEmpty = false;
+      break;
+    }
+  }
+
+  return isEmpty;
+}
+
+/**
+ * This is the helper method to validate the length of Metadata array when it is specified.
+ */
+export function validateMetadata(metadata?: HeaderMatch[]) {
+  const MIN_LENGTH = 1;
+  const MAX_LENGTH = 10;
+
+  if (metadata && (metadata.length < MIN_LENGTH || metadata.length > MAX_LENGTH)) {
+    throw new Error(`Number of metadata provided for matching must be between ${MIN_LENGTH} and ${MAX_LENGTH}. got: ${metadata.length}`);
+  }
+}
+
+/**
+ * This is the helper method to validate at least one of gRPC match type is defined.
+ */
+export function validateGprcMatch(match: GrpcRouteMatch | GrpcGatewayRouteMatch) {
+  if (areMatchPropertiesUndefined(match)) {
+    throw new Error('At least one gRPC match property must be provided.');
+  }
+}
+
+/**
+ * This is the helper method to validate if the path starts with '/'.
+ */
+export function validateStartWith(prefixPathMatch?: string, exactPathMatch?: string,
+  customPrefixRewrite?: string, exactPathRewrite?: string) {
+
+  const pathMatchList = new Map([
+    ['Prefix', prefixPathMatch],
+    ['Exact', exactPathMatch],
+  ]);
+
+  const pathRewriteList = new Map([
+    ['Prefix', customPrefixRewrite],
+    ['Exact', exactPathRewrite],
+  ]);
+
+  for (const key of pathMatchList.keys()) {
+    let value = pathMatchList.get(key);
+
+    if (value && value[0] !== '/') {
+      throw new Error(`${key} Path for the match must start with \'/\', got: ${value}`);
+    }
+  }
+
+  for (const key of pathRewriteList.keys()) {
+    let value = pathRewriteList.get(key);
+
+    if (value && value[0] !== '/') {
+      throw new Error(`${key} Path for the rewrite must start with \'/\', got: ${value}`);
+    }
+  }
+}
```

Comment:
In my opinion, this helper functions adds more complexity than it saves... I would get rid of it.

Reply:
Ok.